### PR TITLE
Lesson 19

### PR DESCRIPTION
**Pull request by haraldF**

### CODE REVIEW

File: slides/lesson19.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   integrity="sha512-GffPMF3RvMeYyc1LWMHtK8EbPv0iNZ8/oTtHPx9/cc2ILxQ+u905qIwdpULaqDkyBKgOaB57QTMg7ztg8Jm2Og=="
+   crossorigin=""></script>
+</head>
+```
+
+---
+
+### Leaflet, cont.
+
+* Next, we need to add a HTML `<div>` element that should contain the map:
```
Comment (sevtapim) [inline slides/lesson19.md]:
```suggestion
* Next, we need to add an HTML `<div>` element in the HTML that should contain the map:
```

File: slides/lesson19.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+```html
+<div id="mapid"></div>
+```
+
+* And we need to give it some width, e.g by setting some CSS in the `<head>` section:
```
Comment (sevtapim) [inline slides/lesson19.md]:
```suggestion
* And we need to give it some width, e.g by setting some CSS in the HTML `<head>` section:
```

File: slides/lesson19.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+```css
+#mapid { height: 180px; }
+```
+
+---
+
```
Comment (sevtapim) [inline slides/lesson19.md:58]: Add a slide saying that rest of the code we are going to add will be written in the JavaScript file

File: 19_leaflet1/main.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// create a new map object, pass the <div> id as parameter:
+let mymap = L.map('mapid');
+// center it on ReDI school, zoom level 13:
+mymap.setView([52.531446, 13.384752], 13);
+
+// create a new layer
```
Comment (bogdanned) [inline 19_leaflet1/main.js]: What about adding an extra slide explaining what a layer is in order to make the transition easier?

File: slides/lesson19.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+The `accessToken` is on the slack channel. If you want to create your own, visit https://www.mapbox.com/studio/account/tokens/
+
+---
+
+### What now?
+
+You should see a map, centered at ReDI school. Using the `setView` function, you can scroll the map to any location.
+
+You can add markers to the map:
```
Comment (bogdanned) [inline slides/lesson19.md:113]: What about splitting this slide in two also. One with setView and the other only with the markers. Shorter slides can allow them to focus better.

File: slides/lesson19.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+marker.bindPopup("<b>Hello world!</b><br>I am a popup.")
+```
+
+Can you add a popup with a fun message to your marker?
+
+---
+
+### Combining APIs
+
+Remember the BVG API?
```
Comment (bogdanned) [inline slides/lesson19.md:138]: Really nice that we are reusing the APIs from past lessons in this one!!!

File: slides/lesson19.md (hunk 3 above)
Comment (haraldF) [reply slides/lesson19.md:58]: I added an extra comment to clarify that the rest is in JavaScript

File: 19_leaflet1/main.js (hunk 4 above)
Comment (haraldF) [reply 19_leaflet1/main.js]: OK, I've added some more comments. Thanks :)

**Pull request merged**